File: Paladins.Api/Paladins.Api/Paladins.Common/Models/PlayerModel.cs
using Paladins.Common.ClientModels.Match;
using Paladins.Common.ClientModels.Player;
using System;
using System.Collections.Generic;
using System.Globalization;
using System.Linq;

namespace Paladins.Common.Models
{
    public class PlayerModel
    {
        public PlayerModel()
        {
            Friends = new List<FriendModel>();
            ChampionStats = new List<PlayerChampionStatsModel>();
            Loadouts = new List<PlayerLoadoutModel>();
            MatchHistories = new List<PlayerMatchHistoryModel>();
        }
        public int PlayerId { get; set; }
        public int PaladinsPlayerId { get; set; }
        public string Name { get; set; }
        public Uri AvatarUrl { get; set; }
        public int AvatarId { get; set; }
        public int MinutesPlayed { get; set; }
        public int HoursPlayed { get; set; }
        public int TotalLeaves { get; set; }
        public int ControllerTier { get; set; }
        public int KbmTier { get; set; }
        public int ConquestTier { get; set; }
        public int TotalWorshippers { get; set; }
        public int TotalXp { get; set; }
        public int TotalWins { get; set; }
        public DateTime LastLoginTimeStamp { get; set; }
        public DateTime AccountCreatedOnTimeStamp { get; set; }
        public int AccountLevel { get; set; }
        public string LoadingFrame { get; set; }
        public int MasteryLevel { get; set; }
        public string PersonalStatusMessage { get; set; }
        public DateTime CreatedOn { get; set; }
        public DateTime LastUpdatedOn { get; set; }
        public bool IsActive { get; set; }
        public List<RankDetailsModel> RankDetailsModel { get; set; }
        public List<FriendModel> Friends { get; set; }
        public List<PlayerChampionStatsModel> ChampionStats { get; set; }
        public List<PlayerLoadoutModel> Loadouts { get; set; }
        public List<PlayerMatchHistoryModel> MatchHistories { get; set; }
        public void PopulateFriends(IList<PlayerFriendsClientModel> friends)
        {
            Friends.AddRange(from f in friends
                             select new FriendModel
                             {
                                 PaladinsPlayerId = Convert.ToInt32(f.PlayerId),
                                 AccountId = Convert.ToInt32(f.AccountId),
                                 Name = f.Name
                             });
        }

        public void PopulateChampionStats(IList<PlayerChampionRanksClientModel> championStats)
        {
            ChampionStats.AddRange(from c in championStats
                                   select new PlayerChampionStatsModel
                                   {
                                       Assists = Convert.ToInt32(c.Assists),
                                       Deaths = Convert.ToInt32(c.Deaths),
                                       Kills = Convert.ToInt32(c.Kills),
                                       Losses = Convert.ToInt32(c.Losses),
                                       PaladinsChampionId = Convert.ToInt32(c.ChampionId),
                                       PlayerId = PlayerId,
                                       Rank = Convert.ToInt32(c.Rank),
                                       Wins = Convert.ToInt32(c.Wins),
                                   });
        }

        public void PopulateLoadouts(IList<PlayerLoadoutsClientModel> loadouts)
        {
            Loadouts.AddRange(from l in loadouts
                              select new PlayerLoadoutModel
                              {
                                  LoadoutName = l.DeckName,
                                  PlayerName = l.PlayerName,
                                  PaladinsPlayerId = Convert.ToInt32(l.PlayerId),
                                  PaladinsChampionId = Convert.ToInt32(l.ChampionId),
                                  LoadoutItems = (from li in l.LoadoutItems
                                                  select new PlayerLoadoutItemModel
                                                  {
                                                      CardName = li.ItemName,
                                                      PaladinsItemId = Convert.ToInt32(li.ItemId),
                                                      PointsAssignedToItem = Convert.ToInt32(li.Points)
                                                  }).ToList()
                              });
        }

        public void PopulateMatchHistory(IList<MatchDetailsClientModel> matchHistories)
        {
            MatchHistories.AddRange(from m in matchHistories
                                    select new PlayerMatchHistoryModel
                                    {
                                        Assists = Convert.ToInt32(m.Assists),
                                        Deaths = Convert.ToInt32(m.Deaths),
                                        Kills = Convert.ToInt32(m.Kills),
                                        MatchResult = m.WinStatus,
                                        //EntryDateTime = DateTime.ParseExact(m.EntryDatetime, "M/d/yyyy h:mm:ss tt", CultureInfo.InvariantCulture),
                                        PaladinsChampionId = Convert.ToInt32(m.ChampionId),
                                        PaladinsPlayerId = Convert.ToInt32(m.PlayerId),
                                        PaladinsMatchId = Convert.ToInt32(m.Match),
                                    });
        }
    }
}

File: Mobile/app/src/main/java/com/fivegearszerochill/paladins/domain/models/player/PlayerModel.kt
package com.fivegearszerochill.paladins.domain.models.player

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import java.util.*

data class PlayerModel(
    @Expose
    @SerializedName("playerId")
    val playerID: Long? = null,

    @Expose
    @SerializedName("paladinsPlayerId")
    val paladinsPlayerID: Long? = null,
    @Expose
    @SerializedName("name")
    val name: String? = null,

    @Expose
    @SerializedName("avatarUrl")
    val avatarURL: String? = null,

    @Expose
    @SerializedName("avatarId")
    val avatarID: Long? = null,
    @Expose
    @SerializedName("minutesPlayed")
    val minutesPlayed: Long? = null,
    @Expose
    @SerializedName("hoursPlayed")
    val hoursPlayed: Long? = null,
    @Expose
    @SerializedName("totalLeaves")
    val totalLeaves: Long? = null,
    @Expose
    @SerializedName("controllerTier")
    val controllerTier: Long? = null,
    @Expose
    @SerializedName("kbmTier")
    val kbmTier: Long? = null,
    @Expose
    @SerializedName("conquestTier")
    val conquestTier: Long? = null,
    @Expose
    @SerializedName("totalWorshippers")
    val totalWorshippers: Long? = null,
    @Expose
    @SerializedName("totalXP")
    val totalXP: Long? = null,
    @Expose
    @SerializedName("totalWins")
    val totalWINS: Long? = null,
    @Expose
    @SerializedName("lastLoginTimeStamp")
    val lastLoginTimeStamp: Date? = null,
    @Expose
    @SerializedName("accountCreatedOnTimeStamp")
    val accountCreatedOnTimeStamp: Date? = null,
    @Expose
    @SerializedName("accountLevel")
    val accountLevel: Long? = null,
    @Expose
    @SerializedName("loadingFrame")
    val loadingFrame: String? = null,
    @Expose
    @SerializedName("masteryLevel")
    val masteryLevel: Long? = null,
    @Expose
    @SerializedName("personalStatusMessage")
    val personalStatusMessage: String? = null,
    @Expose
    @SerializedName("createdOn")
    val createdOn: Date? = null,
    @Expose
    @SerializedName("lastUpdatedOn")
    val lastUpdatedOn: Date? = null,
    @Expose
    @SerializedName("isActive")
    val isActive: Boolean? = null,
    @Expose
    @SerializedName("rankDetailsModel")
    val rankDetailsModel: List<RankDetailsModel>? = null,
    @Expose
    @SerializedName("friends")
    val friends: List<FriendModel>? = null,
    @Expose
    @SerializedName("championStats")
    val championStats: List<ChampionStatModel>? = null,
    @Expose
    @SerializedName("loadouts")
    val loadouts: List<LoadoutModel>? = null,
    @Expose
    @SerializedName("matchHistories")
    val matchHistories: List<MatchHistoryModel>? = null
){
    override fun toString(): String {
        return "PlayerModel(playerID=$playerID, paladinsPlayerID=$paladinsPlayerID, name=$name, avatarURL=$avatarURL, avatarID=$avatarID, minutesPlayed=$minutesPlayed, hoursPlayed=$hoursPlayed, totalLeaves=$totalLeaves, controllerTier=$controllerTier, kbmTier=$kbmTier, conquestTier=$conquestTier, totalWorshippers=$totalWorshippers, totalXP=$totalXP, totalWINS=$totalWINS, lastLoginTimeStamp=$lastLoginTimeStamp, accountCreatedOnTimeStamp=$accountCreatedOnTimeStamp, accountLevel=$accountLevel, loadingFrame=$loadingFrame, masteryLevel=$masteryLevel, personalStatusMessage=$personalStatusMessage, createdOn=$createdOn, lastUpdatedOn=$lastUpdatedOn, isActive=$isActive, rankDetailsModel=$rankDetailsModel, friends=$friends, championStats=$championStats, loadouts=$loadouts, matchHistories=$matchHistories)"
    }
}

